js/data.js

```diff
@@ -7,15 +7,124 @@ var data = {
   nextEntryId: 1
 };
 
-var previousDataJSON = localStorage.getItem('javascript-local-storage');
+var previousDataJSON = localStorage.getItem('code-journal-local-storage');
 
 if (previousDataJSON !== null) {
   data = JSON.parse(previousDataJSON);
 }
 
 function handleStroage(event) {
   var dataJSON = JSON.stringify(data);
-  localStorage.setItem('javascript-local-storage', dataJSON);
+  localStorage.setItem('code-journal-local-storage', dataJSON);
 }
 
 window.addEventListener('beforeunload', handleStroage);
+
+var $entryForm = document.forms[0];
+var $image = document.querySelector('.image');
+
+function handleEntryForm(event) {
+  event.preventDefault();
+  var formValues = {
+    title: $entryForm.elements.title.value,
+    photoUrl: $entryForm.elements.photoUrl.value,
+    notes: $entryForm.elements.notes.value
+  };
+  getUl.prepend(journalEntries(formValues));
+  entriesContainer.className = 'container view';
+  $form.className = 'form-background hidden';
+
+  formValues.entryId = data.nextEntryId;
+  data.nextEntryId++;
+  data.entries.unshift(formValues);
+  $image.setAttribute('src', '/images/placeholder-image-square.jpg');
+  $entryForm.reset();
+  data.view = 'entries';
+}
+
+$entryForm.addEventListener('submit', handleEntryForm);
+
+function journalEntries(entry) {
+  var liElement = document.createElement('li');
+
+  var containerDiv = document.createElement('div');
+  containerDiv.setAttribute('class', 'container');
+  liElement.appendChild(containerDiv);
+
+  var rowDiv = document.createElement('div');
+  rowDiv.setAttribute('class', 'row');
+  containerDiv.appendChild(rowDiv);
+
+  var columnHalfImage = document.createElement('div');
+  columnHalfImage.setAttribute('class', 'column-half');
+  rowDiv.appendChild(columnHalfImage);
+
+  var image = document.createElement('img');
+  image.setAttribute('src', entry.photoUrl);
+  columnHalfImage.appendChild(image);
+
+  var columnHalfText = document.createElement('div');
+  columnHalfText.setAttribute('class', 'column-half');
+  rowDiv.appendChild(columnHalfText);
+
+  var h2Element = document.createElement('h2');
+  h2Element.textContent = entry.title;
+  columnHalfText.appendChild(h2Element);
+
+  var pElement = document.createElement('p');
+  pElement.textContent = entry.notes;
+  columnHalfText.appendChild(pElement);
+
+  return liElement;
+}
+
+var getUl = document.querySelector('ul');
+
+for (var i = 0; i < data.entries.length; i++) {
+  var ulAppend = journalEntries(data.entries[i]);
+  getUl.appendChild(ulAppend);
+}
+
+window.addEventListener('DOMContentLoaded', journalEntries);
```

Comment:
I imagine this is causing an error in your console 😬. DOuble check that all errors are addressed before submitting for review.